### PR TITLE
Add match-constraint options to depends command

### LINKED ISSUE #4661: Composer depends with matched constraint
Idea: Add an option on composer depends to search requirements of packages that match a specific constraint.

Example:

```bash
# Will show all depends that not allow Symfony 3.0+
composer depends symfony/symfony --match-constraint=<3.0
```

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Closes #4661.

Output sample:

```bash
# Shows requirements that can accept Symfony 3.0+
$ composer depends symfony/dependency-injection --match-constraint='~3.0'
sensio/distribution-bundle requires symfony/dependency-injection (~2.3|~3.0)
sllh/iso-codes-validator requires (dev) symfony/dependency-injection (^2.3|^3.0)
symfony/assetic-bundle requires symfony/dependency-injection (~2.3|~3.0)
symfony/monolog-bundle requires symfony/dependency-injection (~2.3|~3.0)
symfony/swiftmailer-bundle requires symfony/dependency-injection (~2.3|~3.0)

# Shows requirements that can NOT accept Symfony 3.0+
$ composer depends symfony/dependency-injection --match-constraint='~3.0' --match-constraint-invert
behat/behat requires symfony/dependency-injection (~2.1)
cocur/slugify requires (dev) symfony/dependency-injection (~2.4)
doctrine/mongodb-odm-bundle requires symfony/dependency-injection (~2.2)
friendsofsymfony/rest-bundle requires (dev) symfony/dependency-injection (~2.3)
nexylan/cloudflare-host-gw requires (dev) symfony/dependency-injection (~2.7)
nexylan/cloudflare requires (dev) symfony/dependency-injection (~2.7)
sonata-project/admin-bundle requires symfony/dependency-injection (~2.3,>=2.3.3)
sonata-project/intl-bundle requires symfony/dependency-injection (~2.2)
```

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Documentation added.

File: src/Composer/Command/DependsCommand.php [hunk 1]
```diff
@@ -43,6 +44,8 @@ protected function configure()
             ->setDefinition(array(
                 new InputArgument('package', InputArgument::REQUIRED, 'Package to inspect'),
                 new InputOption('link-type', '', InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY, 'Link types to show (require, require-dev)', array_keys($this->linkTypes)),
+                new InputOption('match-constraint', '', InputOption::VALUE_OPTIONAL, 'Constraint to match', '*'),
```
Comment (stof) [inline src/Composer/Command/DependsCommand.php]: this should be VALUE_REQUIRED. Passing just ``--match-constraint`` does not make sense

**Comment by stof** [comment]
The help of the command should be updated too, to explain these options

File: src/Composer/Command/DependsCommand.php (hunk 1 above)
Comment (Soullivaneuh) [inline src/Composer/Command/DependsCommand.php]: Indeed.

**Comment by Soullivaneuh** [comment]
Are you talking of the main `setHelp` method?

This is provided without any option so I didn't touch it. Should I?

**Pull request merged**